app/app.rb

```diff
@@ -0,0 +1,86 @@
+ENV["RACK_ENV"] ||= "development"
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require 'encrypted_cookie'
+require_relative 'data_mapper_setup'
+require 'pry'
+require_relative 'helpers'
+
+class Chitter < Sinatra::Base
+  set :public_folder, Proc.new { File.join(root, 'static') }
+  register Sinatra::Flash
+  use Rack::MethodOverride
+  use Rack::Session::EncryptedCookie,
+    secret: 'ad51d97857116860975480f71544061c7b6212d63073ae587c463928c793c7c9'
+
+  DataMapper::Logger.new($stdout, :debug)
+
+  helpers Helpers
+
+  before do
+    Peep.count > 0 ? @peeps = Peep.all(order: :created_at.desc) : @peeps = nil
+  end
+
+  get '/' do
+    erb(:index)
+  end
+
+  get '/users/new' do
+    session[:email] ? @email_res = session[:email] : nil
+    session[:name] ? @name_res = session[:name] : nil
+    session[:username] ? @user_res = session[:username] : nil
```

Comment:
I've no idea what an 'email res' is. Does 'res' have any meaning in your domain? Are you sure it's the best word for this thing?